pkg/gitparse/gitparse_test.go

```diff
@@ -20,6 +29,20 @@ type testCase struct {
 	function func(bool, ParseState, []byte) bool
 }
 
+func TestParsePanicDiff(t *testing.T) {
+	// setup logger
+	logger, sync := log.New("forager/scanner",
+		log.WithConsoleSink(os.Stderr),
+	)
+	defer func() { _ = sync() }()
+	context.SetDefaultLogger(logger)
+
+	ctx := context.Background()
+	parser := NewParser()
+	commitCh := make(chan Commit, 64)
+	parser.FromReader(ctx, io.NopCloser(bytes.NewReader(panicDiff1)), commitCh, false)
```

Comment:
Is the idea of this test that this line *won't* panic? What do you think about using [assert.NotPanics](https://pkg.go.dev/github.com/stretchr/testify/assert#NotPanics) to make that explicit?